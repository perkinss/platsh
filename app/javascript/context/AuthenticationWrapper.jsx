import React, { useEffect, useState, useContext } from 'react';
import User from '../model/user'

export const RootAuthContext = React.createContext();
export default ({ children }) => {
  let anyAuthVal,anyAuthData;
  try {
    anyAuthVal=JSON.parse(window.sessionStorage.getItem('authenticated'));
  }
  catch(err) {
    anyAuthVal=false;
  }
  try {
    anyAuthData = JSON.parse(window.sessionStorage.getItem('authData'))
  }
  catch(err) {
    anyAuthData = null;
  }
  const guestUser = new User({})
  const prevAuth = (anyAuthVal ? anyAuthVal : false);
  const prevAuthData = (anyAuthData ? anyAuthData : null);
  const prevUser = (anyAuthData ? new User(anyAuthData) : guestUser);
  const [authenticated, setAuthenticated] = useState(prevAuth);
  const [authData, setAuthData] = useState(prevAuthData);
  const [currentUser, setCurrentUser] = useState(prevUser);
  useEffect(
    () => {
      window.sessionStorage.setItem('authenticated', JSON.stringify(authenticated));
      window.sessionStorage.setItem('authData', (authData ? JSON.stringify(authData):null));
      setCurrentUser(authData ? new User(authData) : guestUser)
    },
    [authenticated, authData]
  );
  const defaultContext = {
    authenticated, setAuthenticated,
    authData, setAuthData,
    currentUser, setCurrentUser
  };
  return (
    <RootAuthContext.Provider value={defaultContext}>
      {children}
    </RootAuthContext.Provider>
  );
};
