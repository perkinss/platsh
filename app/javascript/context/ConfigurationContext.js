import React from 'react'

export const configuration = {
    sections: [
        {
            id: null,
            courses: [],
            name: '',
            assessment: [
                {
                    type: 'test',
                    name: '',
                    tasks: [
                    {
                        name: '',
                        available_standards: [
                            {
                                id: -1,
                                level: ''
                            }
                        ],
                        competencies: [

                        ]
                    },
                    ]
                }
            ],
            enrollees: [
                1, 2, 3, 22
            ]
        },
    ],

}


const ConfigurationContext =  React.createContext(configuration);

export { ConfigurationContext }
