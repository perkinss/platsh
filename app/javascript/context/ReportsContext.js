import React from 'react'

export const reports = {
    students: [],
    sections: [],
    selected_section: null,
    selected_student: null,
    section_overview_marks: {},
    section_competency_scores: {},
    student_section_marks: {},
    heat_map_colors: []
}


const ReportsContext =  React.createContext(reports);

export { ReportsContext }
