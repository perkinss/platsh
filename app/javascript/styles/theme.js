import {createMuiTheme} from "@material-ui/core/styles";

const theme = createMuiTheme({
    palette: {
        primary: {
            main: '#512da8',
        },
        secondary: {
            main: '#ab47bc',
        },
        type: 'dark'
    },
    slider: {
        selectionColor: '#F76B1C',
        handleFillColor: '#F76B1C',
        thumb: 'secondary'
    }

})

export default theme
