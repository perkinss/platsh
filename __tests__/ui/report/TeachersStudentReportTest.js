import React, { useState } from 'react'
import { createRender, createShallow } from '@material-ui/core/test-utils';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect'
import theme from "../../../app/javascript/styles/theme";
import { ThemeProvider } from '@material-ui/styles'
import TeachersStudentReport from "../../../app/javascript/components/report/TeachersStudentReport";
import {ReportsContext} from "../../../app/javascript/context/ReportsContext";
import {MemoryRouter} from "react-router-dom";

Enzyme.configure({ adapter: new Adapter() });

describe('<TeachersStudentReport />', () => {
    let render, shallow;
    let classes = {
        content: {
            flexGrow: 1,
            padding: theme.spacing(2,0,2,2),
            marginRight: 0,
        },
    }
    let state = {
        selected_student: {
            id: 5, name: "Ed Shearan", unique_id: "edshe@ran.com", created_at: "2019-07-07T05:07:00.505Z", updated_at: "2019-07-07T05:07:00.505Z"
        },
        section_loading: {
            topics: false,
            competency: false,
        },
        student_loading: {
            competency: false,
            topics: false,
            heat_map: false,
            standards: false,
            comments: false,
        },
    }
    let student = {
        id: 5, name: "Ed Shearan", unique_id: "edshe@ran.com", created_at: "2019-07-07T05:07:00.505Z", updated_at: "2019-07-07T05:07:00.505Z"
    }
    let section = {
        name: 'New at the Zoo...',
        courses: [
            {
                title: 'Math 9'
            }
        ]
    }
    beforeAll(() => {
        render = createRender();
        shallow = createShallow();
    })

    it('should match the snapshot', () => {
        const wrapper = shallow(
            <MemoryRouter>
                <ThemeProvider theme={theme}>
                    <ReportsContext.Provider value={{dispatch: () => {}, state: state }} >
                        <TeachersStudentReport classes={classes} student={student} section={section} clearStudent={() => null} />
                    </ReportsContext.Provider>
                </ThemeProvider>
            </MemoryRouter>
                );
        expect(wrapper).not.toBeNull()
        expect(wrapper).toMatchSnapshot()
    });

    it('should have the student details on first render', () => {
        const wrapper = render(
            <MemoryRouter>
                <ThemeProvider theme={theme}>
                    <ReportsContext.Provider value={{dispatch: () => {}, state: state }} >
                        <TeachersStudentReport classes={classes} student={student} section={section}/>
                    </ReportsContext.Provider>
                </ThemeProvider>
            </MemoryRouter>
        );
        let result = wrapper.text()
        expect(result).toContain("Math 9")
        expect(result).toContain("Competency Overview")
        expect(result).toContain("Course Topics Overview")
        expect(result).toContain("Competency Progress Heatmap and Most Recent Mode (MRM)")
    });

})