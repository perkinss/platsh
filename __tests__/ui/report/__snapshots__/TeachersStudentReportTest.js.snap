// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TeachersStudentReport /> should match the snapshot 1`] = `ShallowWrapper {}`;
